Added support for up to 36 menu items

hallem opened the pull request:
Existing support was only for 10 menu items. I added support for 26 more using the alphabet to label them.

hallem committed:
Added support for up to 36 menu items

Existing support was only for 10 menu items. I added support for 26 more using the alphabet to label them.

lechu445 commented on ConsoleMenu/ItemsCollection.cs:
```diff
@@ -97,6 +99,18 @@ internal bool IsSelected(MenuItem menuItem)
+  {
+    int index = ch switch
+    {
+      >= 'a' and <= 'z' => ch - 87,
+      >= 'A' and <= 'Z' => ch - 56,
```
Let's avoid magical constants
```suggestion
      >= 'a' and <= 'z' => ch - 'a' + 10,
      >= 'A' and <= 'Z' => ch - 'A' + 10,
```

hallem replied:
Implemented

lechu445 commented on ConsoleMenu/ItemsCollection.cs:
```diff
@@ -78,12 +78,14 @@ public void SelectPreviousVisibleItem(VisibilityManager visibility)
   public void Select(char ch)
   {
-    this.currentItemIndex = ch - '0';
+    // this.currentItemIndex = ch - '0';
```
Remove commented out code
```suggestion
```

hallem replied:
Implemented

lechu445 commented on ConsoleMenu/ItemsCollection.cs:
```diff
@@ -78,12 +78,14 @@ public void SelectPreviousVisibleItem(VisibilityManager visibility)
   public bool CanSelect(char ch)
   {
-    return ch >= '0' && (ch - '0') < this.menuItems.Count; // is in range 0.._menuItems.Count
+    // return ch >= '0' && (ch - '0') < this.menuItems.Count; // is in range 0.._menuItems.Count
```
Remove commented out code
```suggestion
```

hallem replied:
Implemented

lechu445 reviewed (commented):
Generally, I like the idea. Thanks!

hallem committed:
Implemented PR Feedback

And corrected a spelling error in a method

hallem commented on ConsoleMenu/MenuConfig.cs:
```diff
@@ -53,7 +53,21 @@ internal MenuConfig(MenuConfig config)
+    }
+    else
+    {
+      index = (char)(item.Index + 48);
```
Implemented

hallem commented on ConsoleMenu/MenuConfig.cs:
```diff
@@ -53,7 +53,21 @@ internal MenuConfig(MenuConfig config)
+
+    if (item.Index >= 10)
+    {
+      index = (char)(item.Index + 55);
```
Implemented

hallem commented on ConsoleMenu/MenuConfig.cs:
```diff
@@ -53,7 +53,21 @@ internal MenuConfig(MenuConfig config)
+  {
+    char index;
+
+    if (item.Index >= 10)
```
You're right. I missed that. Maybe the play here is to only use this feature when there are >= 10 and <= 36 items, otherwise just use the default.

The other thing we can try is disabling the ability to use the keyboard to individually select a menu item when there are more than 10 or 36, forcing you to use the arrows to navigate.

What do you thing @lechu445?

hallem replied:
So I fixed the issue where if there are more than 36 items, the indexing reverts back to numerics. But that didn't solve the filtering issue since if you type a letter it's going to try and match it with the actual menu item vs the filter. I'm not sure if there's a way around this.

hallem replied:
Just pushed the latest changes. The only thing that isn't resolved is when the alphabet is enabled, filtering doesn't work. As I mentioned above, clearer lines would need to be drawn between filter mode and keyboard menu item selection.

hallem committed:
PR Feedback & Other stuff

- fixed an issue with 37+ items, will revert back to numbers
- added property to control if the A-Z characters should be used or now
- added property to disable menu item selection via keyboard forcing up/down arrows to be used
- updated test program to test the above features

hallem commented on ConsoleMenu/MenuConfig.cs:
```diff
@@ -53,7 +53,21 @@ internal MenuConfig(MenuConfig config)
 
   /// <summary>default: (item) => Console.Write("[{0}] {1}", item.Index, item.Name)</summary>
-  public Action<MenuItem> WriteItemAction = item => Console.Write("[{0}] {1}", item.Index, item.Name);
+  public Action<MenuItem> WriteItemAction = item =>
```
Added the EnableAlphabet property for this.

lechu445 committed:
Merge branch 'master' into wip

Commit:
Update README

Commit:
Add a remark comment to EnableAlphabet about filtering

lechu445 commented:
Thank you @hallem for the contribution!
